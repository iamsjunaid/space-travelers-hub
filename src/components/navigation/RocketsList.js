import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../../redux/rockets/rocketsSlice';
import Rocket from './Rocket';
import '../../styles/RocketsList.css';

function RocketsList() {
  const { rockets, isLoading } = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  if (isLoading) {
    return <h2 className="rocketsLoading">Loading...</h2>;
  }

  return (
    <ul className="rocketsList">
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.id}
          name={rocket.name}
          description={rocket.description}
          image={rocket.image}
          reserved={rocket.reserved}
        />
      ))}
      ;
    </ul>
  );
}

export default RocketsList;
