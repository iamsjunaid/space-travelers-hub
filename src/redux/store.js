import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    dragons: dragonsReducer,
  },
});

export default store;
