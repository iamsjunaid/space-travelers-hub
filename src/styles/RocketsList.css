.rocketsList {
  display: flex;
  flex-direction: column;
  gap: 1em;
  margin: 1em 2em;
  padding: 0;
}
