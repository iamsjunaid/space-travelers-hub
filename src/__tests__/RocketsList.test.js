import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import React from 'react';
import store from '../redux/store';
import RocketsList from '../components/navigation/RocketsList';

describe('User Interactions test', () => { 
  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketsList />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
