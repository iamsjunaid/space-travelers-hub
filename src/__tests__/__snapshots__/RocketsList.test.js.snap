// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<h2
  className="rocketsLoading"
>
  Loading...
</h2>
`;
